### PR TITLE
Fixed: few syntax errors.  Added: first 4 challenges

### PR DESCRIPTION
**Pull request by halebeau**
Found a few syntax errors in the comments. Got thru the first 4 challenges.  Reviewed the resources

### CODE REVIEW

File: ruby/methods.md [hunk 1]
```diff
@@ -42,18 +42,18 @@ end
 Now we can call the method repeatedly by referencing the name `greeter`.
 
 In IRB:
-```
->greeter
+```ruby
```
Comment (sjproctor) [inline ruby/methods.md:45]: This actually isn't Ruby code. It is from the command line. So it would be "bash"

File: ruby/methods.md [hunk 2]
```diff
@@ -103,14 +103,55 @@ end
...
+  if num.even?
+    "#{num} is even."
+  else
+    "#{num} is odd."
+  end 
+end
+puts is_even 1
+puts is_even 2
+puts is_even 3
+
```
Comment (sjproctor) [inline ruby/methods.md:129]: Based on the prompt we are looking for just a true or false output.

File: ruby/methods.md [hunk 3]
```diff
@@ -103,14 +103,55 @@ end
...
+  if num >= 1 && num <= 10
+    "#{num} is Valid."
+  else
+    "#{num} is Invalid."
+  end
+end
+puts inclusive 7
+puts inclusive 35
+puts inclusive -5
+
```
Comment (sjproctor) [inline ruby/methods.md:141]: Same thing here. The output should match the given prompt.

File: ruby/methods.md [hunk 4]
```diff
@@ -103,14 +103,55 @@ end
...
+#- Create a method that takes in a string and determines if the string is a palindrome.
+word_one = 'civic'
+word_two = 'tool'
+
+def palindrome? string
```
Comment (sjproctor) [inline ruby/methods.md:146]: I'm very interested in where you found the syntax for naming this method. A bunch of the echo students did the exact same thing. It is not a good practice to include a question mark on your method name.

**Review by sjproctor — CHANGES_REQUESTED** [review]

File: ruby/methods.md [hunk 5]
```diff
@@ -104,76 +104,106 @@ end
...
 p sum_these_numbers -15, 30
+```
 
-#- Create a method called is_even, which takes a single integer, and which then returns true if the number is even, and false otherwise.
+ - Create a method called is_even, which takes a single integer, and which then returns true if the number is even, and false otherwise.
+ ```ruby
 def is_even num
   if num.even?
-    "#{num} is even."
+    p true
```
Comment (sjproctor) [inline ruby/methods.md:123]: It is best practice for the `p` to be on the function invocation not on the inner workings of the function.

File: ruby/methods.md [hunk 6]
```diff
@@ -104,76 +104,106 @@ end
...
+
+ - Create a method that takes a number as an argument and prints "Valid" if the number is between 1 and 10 (inclusive) and "Invalid" otherwise.
+```ruby
 def inclusive num
   if num >= 1 && num <= 10
-    "#{num} is Valid."
+    p 'Valid'
   else
-    "#{num} is Invalid."
+    p 'Invalid'
```
Comment (sjproctor) [inline ruby/methods.md:139]: Same thing here, the `p` should be on the invocation.

File: ruby/methods.md [hunk 7]
```diff
@@ -104,76 +104,106 @@ end
...
+ - As a developer, I can create a method that checks for the following rules for a user ID and password:
+   - User ID and password _cannot_ be the same.
+   - User ID and password _must_ be at least six characters long.
+   - Password _must_ contain at least one of: !#$
+   - User ID _cannot_ contain the following characters: !#$ or spaces
+   - Password _cannot_ be the word "password".
+```ruby
+def registration(username, password)
+  if username == password
+    puts 'User ID and password cannot be the same.'
```
Comment (sjproctor) [inline ruby/methods.md:180]: Same thing here with the `puts`. It should be on the invocation.

File: ruby/methods.md [hunk 8]
```diff
@@ -104,76 +104,106 @@ end
...
+  elsif username.length < 6 || password.length < 6
+    puts 'Username and Password must be at least 6 characters'
+  elsif !password.include?('!') && !password.include?('#') && !password.include?('$') 
+    puts 'Password must contain at least one of: !,#,$'
+  elsif username.include?('!') || username.include?('#') || username.include?('$') || username.include?(' ') 
+    puts "Username cannot contain !,#,$,' '"
+  elsif password == 'password' || password == 'Password'
```
Comment (sjproctor) [inline ruby/methods.md:187]: I don't think you will ever get to this line as the one before it will be true if the password is `password`.

**Review by sjproctor — CHANGES_REQUESTED** [review]

File: ruby/methods.md [hunk 9]
```diff
@@ -103,36 +103,147 @@ end
...
+```ruby
+#  - Create a method called sum_these_numbers which takes two integers as an argument and prints their sum to the screen.
+
+def sum_these_numbers(num1, num2)
+  num1 + num2
+end
+p sum_these_numbers 1, 2
+p sum_these_numbers 10, 20
+p sum_these_numbers 100, 200
+p sum_these_numbers -15, 30
```
Comment (sjproctor) [inline ruby/methods.md:116]: It would be a best practice to use parens on the invocation of the method.

**Review by sjproctor — CHANGES_REQUESTED** [review]
Couple small things to clean up.

**Comment by halebeau** [comment]
Changed "puts" to "p".
Added parens to values

**Review by sjproctor — CHANGES_REQUESTED** [review]
I had trouble seeing the changes of the last update pushed on this view. I left comments on the actual commit. The only feedback is that Ruby is really picky about spaces. You will need to remove the space between the function invocation and the parentheses.

**Review by sjproctor — APPROVED** [review]
yay!

**Pull request merged**